Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/self_intersections.h

```diff
@@ -503,7 +503,11 @@ bool does_self_intersect(const FaceRange& face_range,
     CGAL::Emptyset_iterator unused_out;
     internal::self_intersections_impl<ConcurrencyTag>(face_range, tmesh, unused_out, true /*throw*/, np);
   }
+  #if defined(CGAL_LINKED_WITH_TBB)
+  catch(tbb::captured_exception&)
+  #else
   catch(CGAL::internal::Throw_at_output_exception&)
+  #endif
```

Comment:
Even if `CGAL_LINKED_WITH_TBB` is defined, then `does_self_intersect` can be instantiated with `Sequential_tag`. It seems more robust to catch both exceptions:
```suggestion
  #if defined(CGAL_LINKED_WITH_TBB)
  catch(tbb::captured_exception&) {
    return true;
  }
  #else
  catch(CGAL::internal::Throw_at_output_exception&)
  #endif
```
but even then, if would be a lot more robust to verify that the exception caught be TBB is of type `CGAL::internal::Throw_at_output_exception`, maybe with something like:
```c++
  #if defined(CGAL_LINKED_WITH_TBB)
  catch(tbb::captured_exception& e) {
    if(e.name() == typeid(CGAL::internal::Throw_at_output_exception).name())
      return true;
    else
      throw;
  }
  #else
```

Reply:
Actually, you are right. This is something I missed.

Reply:
In fact, since both `TBB` and `Throw_at_output_exception` are derived from the `std::exception` and we do not really need to handle them but rather return upon their arrival, I changed to `std::exception`. This seems cleaner and more general.